cmd/internal/cli/build.go

```diff
@@ -224,3 +225,53 @@ func makeDockerCredentials(cmd *cobra.Command) (authConf *ocitypes.DockerAuthCon
 
 	return
 }
+
+// remote builds need to fail if we cannot resolve remote URLS
+func handleRemoteBuildFlags(cmd *cobra.Command) {
+	// if we can load config and if default endpoint is set, use that
+	// otherwise fall back on regular authtoken and URI behavior
+	e, err := sylabsRemote(remoteConfig)
+	if err == nil {
+		authToken = e.Token
+		if !cmd.Flags().Lookup("builder").Changed {
+			uri, err := e.GetServiceURI("builder")
+			if err != nil {
+				sylog.Fatalf("Unable to get build service URI: %v", err)
+			}
+			builderURL = uri
+		}
+		if !cmd.Flags().Lookup("library").Changed {
+			uri, err := e.GetServiceURI("library")
+			if err != nil {
+				sylog.Fatalf("Unable to get library service URI: %v", err)
+			}
+			libraryURL = uri
+		}
+	} else if err == scs.ErrNoDefault {
+		sylog.Warningf("No default remote in use, falling back to CLI defaults")
+	} else {
+		sylog.Fatalf("Unable to load remote configuration: %v", err)
+	}
+}
+
+// standard builds should just warn and fall back to CLI default if we cannot resolve library URL
+func handleBuildFlags(cmd *cobra.Command) {
+	// if we can load config and if default endpoint is set, use that
+	// otherwise fall back on regular authtoken and URI behavior
+	e, err := sylabsRemote(remoteConfig)
+	if err == nil {
```

Comment:
Here same comment as in `handleRemoteBuildFlags`. I think in general the fewer and shallower conditionals we use the better off we will be in terms of readability of code